.github/workflows/build-publish-mcr.yaml

```diff
@@ -0,0 +1,49 @@
+name: Build and publish to ACR
+
+on:
+  push:
+    branches:
+      - main  # Or your release branch
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Azure login via OIDC
+      uses: azure/login@v1
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+    - name: Install tools
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y curl jq make
+        curl -LO https://github.com/google/ko/releases/download/v0.13.0/ko-linux-amd64
+        chmod +x ko-linux-amd64 && sudo mv ko-linux-amd64 /usr/local/bin/ko
+        curl -LO https://get.helm.sh/helm-v3.12.0-linux-amd64.tar.gz
+        tar -zxvf helm-v3.12.0-linux-amd64.tar.gz && sudo mv linux-amd64/helm /usr/local/bin/helm
+        curl -sSfL https://github.com/mikefarah/yq/releases/download/v4.43.1/yq_linux_amd64 -o /usr/local/bin/yq
+        chmod +x /usr/local/bin/yq
+        curl -LO https://github.com/sigstore/cosign/releases/download/v2.0.0/cosign-linux-amd64
+        chmod +x cosign-linux-amd64 && sudo mv cosign-linux-amd64 /usr/local/bin/cosign
+
+    - name: Run release
+      env:
+        RELEASE_ACR: AKSMCRImagesCommon
```

Comment:
Leave soem bread crumbs about who owns this if it breaks? Wasa  amystery to find it feels like